### DISCUSSION TITLE
document.getElementById still null when navigate to other page

### DISCUSSION THREAD

**Discussion by ozonerik** [livewire/livewire#8138] [1. Help] [answered]
![image](https://github.com/livewire/livewire/assets/49719997/a1e4dd12-bb8d-4b97-b9bc-ca65a86df78e)

![image](https://github.com/livewire/livewire/assets/49719997/54f1b851-6256-4e8d-a228-b32e3c86d2b6)

when goto add page input id cant read by getElementById, how to re initialized conitainer id every page
![image](https://github.com/livewire/livewire/assets/49719997/a0a87919-6ddf-49aa-9b78-7f868453ec7e)

**Comment by danie-ramdhani** [comment, marked as answer]
in livewire, we have `@assets` to push css and script and `@script` to push javascript for alternative  `@push` if you're using `wire:navigate`.
if you want to run a javascript everytime the page is navigated, use `livewire:navigated` not `livewire:initialized`.

made a example for you: https://wirebox.app/b/4oprw

> next time if you have problem / need help, please provide the code, not screenshots and the problem you have and describe the problem you have.

**Reply by ozonerik** [reply]
thanks for your reply, i will try it. :)

**Reply by ozonerik** [reply]
i was try it, but still null

this is my view code: [https://github.com/ozonerik/eSawah11/blob/main/resources/views/livewire/backend/sawahs.blade.php](https://github.com/ozonerik/eSawah11/blob/main/resources/views/livewire/backend/sawahs.blade.php)

ant this is my controller code:  [https://github.com/ozonerik/eSawah11/blob/main/app/Livewire/Backend/Sawahs.php](https://github.com/ozonerik/eSawah11/blob/main/app/Livewire/Backend/Sawahs.php)

i'm confused whats the problem, coz if i try you simple code its work..

**Reply by danie-ramdhani** [reply]
use `@script` https://livewire.laravel.com/docs/javascript#executing-scripts

**Reply by ozonerik** [reply]
![image](https://github.com/livewire/livewire/assets/49719997/75736215-05e6-476b-a5ba-ea8504d51b79)
![image](https://github.com/livewire/livewire/assets/49719997/b617d658-51c1-4227-8249-d0b5c2ff386a)

i was change code, but still not working when i fisrt click, but when second click, id lokasiadd can read.
